### ISSUE TITLE
We should port the VSO post-search filtering code to work with Fido

### ISSUE THREAD

**Issue by Cadair** [sunpy/sunpy#3733]
In VSO you can use attrs to filter the search results, we should port this to Fido.

**Comment by twentyse7en** [comment]
Tagging the issue could be very helpful

**Comment by aribalam** [comment]
I think we should add the code for filtering results similar to what has been implemented in `sunpy/net/vso/attrs.py`. But this cannot be done unless #3714 is merged.
Any thoughts on this @nabobalis @Cadair ?

**Comment by nabobalis** [comment]
I don't see why it has to wait for that to be merged in.

**Comment by aribalam** [comment]
The current `filter_result` code in `vso/attrs.py` uses `MultiMethod` which will be deprecated after this PR. Also, we can make use of the newly created `seconddispatch` function.

**Comment by Cadair** [comment]
Yeah probably would be easier to base it off that branch, but you could checkout those commits and start working on it before it's merged. (I hope to merge it today anyway)

**Comment by aribalam** [comment]
@Cadair @nabobalis It might be really complicated to filter out queries as the data format varies hugely depending on the client. Moreover there are some issues regarding data returned by the VSOClient (as mentioned in #3833). The `QueryResponseBlock` list returned is unhashable and cannot be converted to a set which is necessary to remove duplicate queries. The only way I could think of implementing this if every client implements its own way of filtering out data (similar to what is done with attrubute walker). Any suggestions on this?

**Comment by nabobalis** [comment]
Could we standardize the data format from each client?

**Comment by abhijeetmanhas** [comment]
@nabobalis By standardization do you mean that all columns shoud be there in Response, and we could specify through query what columns to display.
Or it is more like generalizing the functions within client classes like have attempted to standardize `_get_time_for_url` here #3863?

**Comment by abhijeetmanhas** [comment]
After #4011 is merged, I would like to work on it. Specifically we need to have filter for dataretreiver client responses and jsoc responses, similar to vso one. And then implement a `search()` method in `UnifiedResponse` which uses existing filters and merge responses from all clients.

**Comment by ankitkhushwaha** [comment]
@nabobalis 
i think we can now use attrs to filter the search results with Fido and
this issue already being fixed ??

**Comment by nabobalis** [comment]
> [<img alt="" width="16" height="16" src="https://avatars.githubusercontent.com/u/1392107?s=64&amp;u=a675cc501ce9d5ca6bb0b86cddb03f93f51abb2d&amp;v=4">@nabobalis](https://github.com/nabobalis?rgh-link-date=2025-02-02T11%3A51%3A08.000Z) i think we can now use attrs to filter the search results with Fido and this issue already being fixed ??

Can we? If so, I can close this issue but we should make sure there is a how-to-guide or an example on how to do it, if there is not one already.

**Comment by ankitkhushwaha** [comment]
in this [example](https://docs.sunpy.org/en/stable/how_to/search_multiple_wavelengths.html) we can  use attrs to filter the search results, with Fido
